### ISSUE TITLE
TypeError: Ellipsis is not a dataclass and cannot be turned into one.

### ISSUE THREAD

**Issue by heckad** [lovasoa/marshmallow_dataclass#63]
```py
from dataclasses import field
from decimal import Decimal
from typing import Tuple

from marshmallow_dataclass import dataclass


@dataclass(frozen=True)
class Signal:
    fields: Tuple[Decimal, ...] = field(default_factory=list)
```

This ellipsis means that all values in fields have Decimal type.

**Comment by lovasoa** [comment]
Unfortunately, marshmallow does not seem to support variable length `Tuple`.
As per [the documentation](https://marshmallow.readthedocs.io/en/stable/api_reference.html#marshmallow.fields.Tuple): 
> *class marshmallow.fields.Tuple*: A tuple field, composed of a **fixed number** of other Field classes or instances

@sloria : Any opinion ?

**Comment by heckad** [comment]
How to make frozen dataclass with a list?

**Comment by lovasoa** [comment]
> How to make frozen dataclass with a list?

I think your best bet for now is to create your own `VariableLengthTuple` field. See [marshmallow's documentation on custom fields](https://marshmallow.readthedocs.io/en/stable/custom_fields.html).

Then, you can have:


```py
@dataclass(frozen=True)
class Signal:
    fields: Tuple[Decimal, ...] = field(
        default_factory=tuple,
        metadata = {"marshmallow_field":VariableLengthTuple()})
```

**Comment by sloria** [comment]
desert (a fork of marshmallow_dataclass) added support for variadic tuples. The PR would be worth evalutating: https://github.com/python-desert/desert/pull/17

**Comment by lovasoa** [comment]
@sloria : Would you consider adding VariadicTuple to marshmallow itself?

**Comment by mikeholler** [comment]
We just ran into this as well. Support for variadic tuples would be cool but in the end the workaround is pretty simple.

**Comment by deansg** [comment]
I'm not sure when/how, but it seems to me like this is now resolved (I tried with version 8.5.10). Can anyone else confirm?

**Comment by dairiki** [comment]
> I'm not sure when/how, but it seems to me like this is now resolved (I tried with version 8.5.10). Can anyone else confirm?

This was indeed fixed in #221, which is included as of release 8.5.10.
(I was working on fixing #181 and independently noticed and fixed this issue as well, unaware of this ticket.)
